python_on_whales/client_config.py

```diff
@@ -37,10 +37,23 @@ def add_flag(self, name: str, value: bool):
         if value:
             self.append(name)
 
-    def add_args_list(self, arg_name: str, list_values: list):
-        for value in to_list(list_values):
+    def add_args_iterable(self, arg_name: str, values: Iterable[Any]):
+        for value in values:
             self.extend([arg_name, value])
```

Comment:
I don't think this is needed. It is indeed faster than the other function and more efficient, but as long as the performance drop is not noticeable, let's not add more code to this codebase.

I had measure this a while algo, and most of the time spent in python-on-whales is actually spent calling subprocess. I double such an optimisation will have an impact

Reply:
Sure, but it's not about performance, it's a correctness thing. I think this kind of improvement is highly valuable, it more clearly (and *correctly*) conveys the types being passed around rather than a general "best-effort" type of approach. If a single value is accidentally passed to this method it would cause an error, which is desirable. This kind of thing would help avoid problems like https://github.com/gabrieldemarmiesse/python-on-whales/issues/466.

Reply:
I see what you mean. I would believe that would be necessary for public functions, but since it's not public, I see it as less useful. But anyway, I don't feel that strongly about this, if you see its value when doing your refactoring, it's good enough for me. I'll approve it :)